exo-go/src/terraform/index.go

```diff
@@ -11,33 +11,33 @@ import (
 )
 
 // GenerateFile generates the main terraform file given application and service configuration
-func GenerateFile(appConfig types.AppConfig, serviceConfigs map[string]types.ServiceConfig, appDir, homeDir, terraformDir string) error {
-	fileData, err := Generate(appConfig, serviceConfigs, appDir, homeDir)
+func GenerateFile(config types.TerraformConfig) error {
+	fileData, err := Generate(config)
 	if err != nil {
 		return err
 	}
-	err = WriteTerraformFile(fileData, terraformDir)
+	err = WriteTerraformFile(fileData, config.TerraformDir)
 	return err
 }
 
 // Generate generates the contents of the main terraform file given application and service configuration
-func Generate(appConfig types.AppConfig, serviceConfigs map[string]types.ServiceConfig, appDir, homeDir string) (string, error) {
+func Generate(config types.TerraformConfig) (string, error) {
 	fileData := []string{}
 
-	moduleData, err := generateAwsModule(appConfig)
+	moduleData, err := generateAwsModule(config.AppConfig, config.RemoteBucket, config.LockTable, config.Region)
```

Comment:
Thoughts on passing the entires config object into this method? Same for `generateDependencyModules`